src/openrct2/ride/Vehicle.cpp

```diff
@@ -8270,20 +8261,19 @@ bool Vehicle::UpdateTrackMotionForwards(rct_ride_entry_vehicle* vehicleEntry, Ri
     moveInfo = vehicle_get_move_info(TrackSubposition, track_type, track_progress);
     trackType = GetTrackType();
     {
-        int16_t curX = TrackLocation.x + moveInfo->x;
-        int16_t curY = TrackLocation.y + moveInfo->y;
-        int16_t curZ = TrackLocation.z + moveInfo->z + RideData5[curRide->type].z_offset;
+        CoordsXYZ loc = { TrackLocation.x + moveInfo->x, TrackLocation.y + moveInfo->y,
+                          TrackLocation.z + moveInfo->z + RideData5[curRide->type].z_offset };
```

Comment:
```suggestion
        CoordsXYZ loc = TrackLocation + CoordsXYZ{ moveInfo->x, moveInfo->y, moveInfo->z + RideData5[curRide->type].z_offset };
```
With proper formatting.